conn_linux.go

```diff
@@ -61,11 +58,11 @@ func dial(family int, config *Config) (*conn, uint32, error) {
 
 	sock, err := newSysSocket(config)
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, os.NewSyscallError("netlink-sys-socket", err)
```

Comment:
I don't think this is quite right. This call to `newSysSocket` only fails if this call fails:
https://github.com/mdlayher/netlink/blob/master/conn_linux.go#L597

That, in itself, can only fail here:
https://github.com/mdlayher/netlink/blob/master/netns_linux.go#L18

And here:
https://github.com/mdlayher/netlink/blob/master/netns_linux.go#L26

In `getThreadNetNS`, we already get a `SyscallError`, from `os.Open`, so there is nothing to do there.

In `setThreadNetNS`, we should wrap the error in `os.NewSyscallError("setns", ...)`.

Then, the error should just be bubbled up without further decoration from this site. This should be documented in case it changes in the future, and we _do_ end up needing more decoration; but that seems unlikely.

Reply:
Agreed, and will update. To clarify, where should this be documented?

Reply:
Right here would be fine, I think.

Reply:
Do you think I should do `_ = err.(*os.SyscallError)` to ensure the contract isn't violated?

Reply:
It appears this won't work; note that os.Open says it returns a PathError: https://golang.org/pkg/os/#Open.

Thoughts?

Reply:
That wouldn't be exactly right either, because https://github.com/mdlayher/netlink/blob/master/netns_linux.go#L18 is probably an `*os.PathError` with an inner `*os.SyscallError`.

Reply:
Understood, I'll just document it and state that bare syscall errors should be wrapped.